airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/api.py

```diff
@@ -97,26 +98,31 @@ def call(
 class API:
     """Simple wrapper around Facebook API"""
 
-    def __init__(self, account_id: str, access_token: str):
-        self._account_id = account_id
+    def __init__(self, config: SourceFacebookMarketingConfig):
+
+        self.__config = config
+
         # design flaw in MyFacebookAdsApi requires such strange set of new default api instance
-        self.api = MyFacebookAdsApi.init(access_token=access_token, crash_log=False)
+        self.api = MyFacebookAdsApi.init(access_token=self.__config.access_token, crash_log=False)
         FacebookAdsApi.set_default_api(self.api)
 
-    @cached_property
-    def account(self) -> AdAccount:
-        """Find current account"""
-        return self._find_account(self._account_id)
+    @cached_property_with_ttl(3600)
+    def accounts(self) -> List[Type[AdAccount]]:
+        """Find current accounts"""
+        return self._find_accounts()
 
-    @staticmethod
-    def _find_account(account_id: str) -> AdAccount:
-        """Actual implementation of find account"""
+    def _find_accounts(self) -> List[Type[AdAccount]]:
+        """Actual implementation of find accounts"""
         try:
-            accounts = fb_user.User(fbid="me").get_ad_accounts()
-            for account in accounts:
-                if account["account_id"] == account_id:
-                    return account
+            accounts_found = list(fb_user.User(fbid="me").get_ad_accounts())
+            if self.__config.account_selection_strategy_is_subset:
+                account_ids = self.__config.account_ids
+                accounts_found = list(
+                    filter(
+                        lambda x: not account_ids or x["account_id"] in account_ids,
+                        accounts_found
+                    )
+                )
+            return accounts_found
         except FacebookRequestError as exc:
             raise FacebookAPIException(f"Error: {exc.api_error_code()}, {exc.api_error_message()}") from exc
-
```

Comment:
what will happen if we can't find some accounts?

Reply:
- If the api raises an exception, that exception is forwarded, same as before. 
- If you are using the subset strategy and some of the specified accounts are not found, those accounts are not included in list. 
- If you are using the subset strategy and all of the accounts are not found, an empty list of accounts is returned. 
- If you are using the all strategy, all found accounts are returned.

I'm not sure I'm understanding the question, am I missing something?

Reply:
Not finding some accounts raised an Exception. Now that we handle one or many accounts, I'm not sure what the behaviour should be. If I provide 5 accounts and I don't have access to one of them, do I want to raise an exception and break the flow, log a warning one of the accounts is missing or simply let the list have 4 accounts? I'm not sure what is the cleanest behaviour.

Reply:
@mlavoie-sm360 
I think the check should succeed only if all specified accounts are accessible.

Reply:
@keu, thanks for your input. Is it all the same to you if that check is done in `source.py` under `def check_connection()` rather than here? At the moment, `check_connection` fails if:
- strategy is `subset` and at least one account is missing from the list
- strategy is `all` and no accounts are available
- strategy selected is unsupported

Reply:
@mlavoie-sm360 I agree with the above if the check is done in the `source.py`, my only concern is that how the user will know that some accounts are not accessible anymore? How about we will have a warning for each missing account in `_find_account` method?

Reply:
@keu, how about this?